code/game/objects/items/storage/backpack.dm

```diff
@@ -89,6 +91,22 @@
 	user.visible_message("<span class='suicide'>[user] places [src] over [user.p_their()] head and pulls it tight! It looks like [user.p_they()] [user.p_are()]n't in the Christmas spirit...</span>")
 	return (OXYLOSS)
 
+/obj/item/storage/backpack/santabag/proc/regenerate_presents()
+	addtimer(CALLBACK(src, .proc/regenerate_presents), rand(30 SECONDS, 60 SECONDS))
+
+	var/mob/M = get(loc, /mob)

```

Comment:
I've never seen this proc before but if you're checking type then just put `loc`, no need for the call.

Reply:
This is essentially in case you get weird things like a backpack inside another backpack. It follows loc until it either hits that type, or gets to the bottom of the well.